### DISCUSSION TITLE
Execute dagstermill op in Python

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#8465] [Q&A]
Hi team, I am trying to execute a job that has notebook op using dagstermill. I can execute this from dagit successfully but using the python API, I am getting this error `dagstermill.errors.DagstermillError: Can't execute a dagstermill op from a job that is not reconstructable. Use the reconstructable() function if executing from python`  what does this mean? Thank you a lot

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C01U954MEER/p1648572523517659?thread_ts=1648572523.517659&cid=C01U954MEER).*

**Comment by yuhan** [comment]
A dagstermill op will be executed in the notebook process which is separate from other op processes, so the error suggests that dagstermill ops need to be run inside a reconstructable job, because when the job is running cross process boundaries, Dagster needs to know how to reconstruct the job on the other side of the process boundary. To make it work, you can wrap the job with [`reconstructable`](https://docs.dagster.io/_apidocs/jobs#reconstructable-jobs), such as:
```python
from dagster import job, reconstructable

def define_my_job():
    @job
    def my_job():
        ...

    return my_job

reconstructable(define_my_job)
```


This is also similar to the question asked in https://github.com/dagster-io/dagster/discussions/8467
